src/java/org/apache/cassandra/db/streaming/CassandraStreamHeader.java

```diff
@@ -86,6 +86,10 @@ private CassandraStreamHeader(Builder builder)
         isEntireSSTable = builder.isEntireSSTable;
         componentManifest = builder.componentManifest;
         firstKey = builder.firstKey;
+
+        if (compressionMetadata != null && compressionInfo == null)
```

Comment:
we should go with the approach in https://issues.apache.org/jira/browse/CASSANDRA-15861 that keeps this lazy and fixes `calculateSize` to deal with this case